src/bytes_mut.rs

```diff
@@ -1732,6 +1733,35 @@ unsafe fn shared_v_to_vec(data: &AtomicPtr<()>, ptr: *const u8, len: usize) -> V
     }
 }
 
+unsafe fn shared_v_to_mut(data: &AtomicPtr<()>, ptr: *const u8, len: usize) -> BytesMut {
+    let shared: *mut Shared = data.load(Ordering::Relaxed).cast();
+
+    if (*shared).is_unique() {
+        let shared = &mut *shared;
+
+        // The capacity is always the original capacity of the buffer
+        // minus the offset from the start of the buffer
+        let v = &mut shared.vec;
+        let v_capacity = v.capacity();
+        let v_ptr = v.as_mut_ptr();
+        let offset = offset_from(ptr as *mut u8, v_ptr);
+        let cap = v_capacity.checked_sub(offset).unwrap_or(len);
```

Comment:
What's the logic of this line? When will the subtraction underflow?

Reply:
I am pretty sure it cant overflow but I wanted to be safe. I can put a debug assert instead and just do a normal sub.

Reply:
Normal subtraction already has a debug assertion.